cloud/scope/cluster.go

```diff
@@ -37,12 +39,12 @@ type ClusterScopeParams struct {
 	Logger       logr.Logger
 	Cluster      *clusterv1.Cluster
 	AzureCluster *infrav1.AzureCluster
-	Context      context.Context
+	NewSKUCache  resourceskus.NewCacheFunc
 }
 
 // NewClusterScope creates a new Scope from the supplied parameters.
 // This is meant to be called for each reconcile iteration.
-func NewClusterScope(params ClusterScopeParams) (*ClusterScope, error) {
+func NewClusterScope(ctx context.Context, params ClusterScopeParams) (*ClusterScope, error) {
```

Comment:
@alexeldeib any thoughts on this one?

Reply:
fixed, honestly it's a bit uglier because background refresh isn't here yet

still no concurrent access so no mutex required even for data mutation